File: tools/bench/answers/go_haiku/s3/06_stats.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{4, 8, 15, 16, 23}
	min := nums[0]
	max := nums[0]
	sum := 0
	for _, n := range nums {
		if n < min {
			min = n
		}
		if n > max {
			max = n
		}
		sum += n
	}
	mean := float64(sum) / float64(len(nums))
	fmt.Println(min)
	fmt.Println(max)
	fmt.Println(mean)
}

File: tools/bench/answers/rust_haiku/s2/06_stats.rs
fn main() {
    let nums = [4, 8, 15, 16, 23];

    let min = nums.iter().min().unwrap();
    let max = nums.iter().max().unwrap();
    let mean = nums.iter().sum::<i32>() as f64 / nums.len() as f64;

    println!("{}", min);
    println!("{}", max);
    println!("{}", mean);
}
